src/sentry/monitors/clock_tasks/check_missed.py

```diff
@@ -0,0 +1,157 @@
+from __future__ import annotations
+
+import logging
+from datetime import datetime
+
+from arroyo.backends.kafka import KafkaPayload
+from sentry_kafka_schemas.schema_types.monitors_clock_tasks_v1 import MarkMissing
+
+from sentry.constants import ObjectStatus
+from sentry.monitors.logic.mark_failed import mark_failed
+from sentry.monitors.models import (
+    CheckInStatus,
+    MonitorCheckIn,
+    MonitorEnvironment,
+    MonitorStatus,
+    MonitorType,
+)
+from sentry.monitors.schedule import get_prev_schedule
+from sentry.utils import metrics
+
+from .producer import MONITORS_CLOCK_TASKS_CODEC, produce_task
+
+logger = logging.getLogger(__name__)
+
+
+# This is the MAXIMUM number of MONITOR this job will check.
+#
+# NOTE: We should keep an eye on this as we have more and more usage of
+# monitors the larger the number of checkins to check will exist.
+MONITOR_LIMIT = 10_000
+
+
+def dispatch_check_missing(ts: datetime):
+    """
+    Given a clock tick timestamp determine which monitor environments are past
+    their next_checkin_latest, indicating they haven't checked-in when they
+    should have
+
+    This will dispatch MarkMissing messages into monitors-clock-tasks.
+    """
+    qs = (
+        # Monitors that have reached the latest checkin time
+        MonitorEnvironment.objects.filter(
+            monitor__type__in=[MonitorType.CRON_JOB],
+            next_checkin_latest__lte=ts,
+        )
+        .exclude(
+            status__in=[
+                MonitorStatus.DISABLED,
+                MonitorStatus.PENDING_DELETION,
+                MonitorStatus.DELETION_IN_PROGRESS,
+            ]
+        )
+        .exclude(
+            monitor__status__in=[
+                ObjectStatus.DISABLED,
+                ObjectStatus.PENDING_DELETION,
+                ObjectStatus.DELETION_IN_PROGRESS,
+            ],
+        )
+        .exclude(
+            monitor__is_muted=True,  # Temporary fix until we can move out of celery or reduce load
+        )
+        .exclude(
+            is_muted=True,  # Temporary fix until we can move out of celery or reduce load
```

Comment:
Should we remove the comment or something?

Reply:
I'll adjust the comment to be a TODO